crates/common/src/shell.rs

```diff
@@ -0,0 +1,242 @@
+//! Utility functions for writing to [`stdout`](std::io::stdout) and [`stderr`](std::io::stderr).
+//!
+//! Simplified adaptation from https://github.com/foundry-rs/foundry/blob/master/crates/common/src/io/macros.rs.
+
+use std::io::{self, Write};
+use std::sync::{Mutex, OnceLock};
+
+/// The output mode: either normal output or completely quiet.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum OutputMode {
+    Normal,
+    Quiet,
+}
+
+/// Choices for whether to use colored output.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum ColorChoice {
+    Auto,
+    Always,
+    Never,
+}
+
+/// A simple shell abstraction.
+///
+/// We only track a verbosity level, an output mode,
+/// and a color choice.
+#[derive(Debug)]
+pub struct Shell {
+    /// Verbosity level (currently unused, but will be in #577)
+    pub verbosity: u8,
+    /// Whether to output anything at all.
+    pub output_mode: OutputMode,
+    /// Whether to use colors.
+    pub color_choice: ColorChoice,
+}
+
+impl Shell {
+    /// Create a new shell with default settings.
+    pub fn new() -> Self {
+        Self {
+            verbosity: 0,
+            output_mode: OutputMode::Normal,
+            color_choice: ColorChoice::Auto,
+        }
+    }
+
+    /// Print a string to stdout.
+    pub fn print_out(&mut self, msg: &str) -> io::Result<()> {
+        if self.output_mode == OutputMode::Quiet {
+            return Ok(());
+        }
+        let stdout = io::stdout();
```

Comment:
IMHO there is value in using [anstream](https://docs.rs/anstream/0.6.18/anstream/) just like anvil does. It is already in our dependency tree because of `clap` so little reason not to use it. It also comes with [anstyle](https://docs.rs/crate/anstyle/latest) so we could eventually get rid of `colored`.

AFAIU this is would be most important for Windows which we don't support right now. Nevertheless, there is little reason to stick with `std::io::stdout`. Can be done in a separate PR though.